{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "import pandas as pd \n",
    "from pandas.io.json import json_normalize\n",
    "\n",
    "\n",
    "url = \"https://raw.githubusercontent.com/OBOFoundry/OBOFoundry.github.io/master/registry/ontologies.jsonld\"\n",
    "obo = requests.get(url)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "obo_json = obo.json()['ontologies']\n",
    "\n",
    "obo_df = json_normalize(obo_json)\n",
    "obo_df.head()\n",
    "\n",
    "\n",
    "obo_df_active = obo_df[obo_df['activity_status'] == 'active']\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['DO wiki', 'activity_status', 'alternatePrefix', 'alternativePrefix',\n",
       "       'biosharing', 'browsers', 'build.checkout', 'build.email_cc',\n",
       "       'build.infallible', 'build.insert_ontology_id', 'build.method',\n",
       "       'build.notes', 'build.oort_args', 'build.path', 'build.source_url',\n",
       "       'build.system', 'canonical', 'contact.contact', 'contact.email',\n",
       "       'contact.github', 'contact.label', 'createdWith', 'dependencies',\n",
       "       'depicted_by', 'description', 'development.id_policy', 'documentation',\n",
       "       'domain', 'exampleClass', 'facebook', 'funded_by', 'google_plus',\n",
       "       'homepage', 'id', 'in_foundry', 'in_foundry_order',\n",
       "       'integration_server', 'is_obsolete', 'jobs', 'label', 'layout',\n",
       "       'license.label', 'license.logo', 'license.url', 'mailing_list',\n",
       "       'ontology_purl', 'page', 'preferredPrefix', 'products', 'publications',\n",
       "       'redirects', 'releases', 'replaced_by', 'repository', 'source',\n",
       "       'source_url', 'taxon.id', 'taxon.label', 'termgenie', 'title',\n",
       "       'tracker', 'twitter', 'type', 'usages', 'used_by', 'validate',\n",
       "       'wasDerivedFrom', 'wikidata_template'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "obo_df_active.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "obo_df_active_selection = obo_df_active[['title', 'description', 'taxon.id','homepage']]\n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "1\n",
      "2\n",
      "failed: \n",
      "Human Disease Ontology\n",
      "3\n",
      "4\n",
      "5\n",
      "failed: \n",
      "Phenotype And Trait Ontology\n",
      "6\n",
      "7\n",
      "failed: \n",
      "PRotein Ontology (PRO)\n",
      "8\n",
      "9\n",
      "failed: \n",
      "Zebrafish anatomy and development ontology\n",
      "10\n",
      "failed: \n",
      "Anatomical Entity Ontology\n",
      "11\n",
      "failed: \n",
      "Agronomy Ontology\n",
      "12\n",
      "failed: \n",
      "Ascomycete phenotype ontology\n",
      "13\n",
      "14\n",
      "failed: \n",
      "Antibiotic Resistance Ontology\n",
      "15\n",
      "failed: \n",
      "Beta Cell Genomics Ontology\n",
      "16\n"
     ]
    }
   ],
   "source": [
    "from SPARQLWrapper import SPARQLWrapper, JSON\n",
    "\n",
    "def get_id(name):\n",
    "    \n",
    "    NAME_LOWERCASE = name.lower()\n",
    "    endpoint_url = \"https://query.wikidata.org/sparql\"\n",
    "\n",
    "    query = \"\"\"SELECT ?item ?itemLabel\n",
    "            WHERE { \n",
    "              ?item p:P31/ps:P31/wdt:P279* wd:Q324254.\n",
    "              ?item rdfs:label ?itemLabel. \n",
    "              FILTER(CONTAINS(LCASE(?itemLabel),\"\"\" +'\"' + NAME_LOWERCASE + '\"' + \"\"\" @en)). \n",
    "            } limit 10\n",
    "            \"\"\"\n",
    "\n",
    "    def get_results(endpoint_url, query):\n",
    "        sparql = SPARQLWrapper(endpoint_url)\n",
    "        sparql.setQuery(query)\n",
    "        sparql.setReturnFormat(JSON)\n",
    "        return sparql.query().convert()\n",
    "\n",
    "\n",
    "    results = get_results(endpoint_url, query)\n",
    "\n",
    "    wikidata_id = results['results']['bindings'][0]['item']['value'].split('/')[4]\n",
    "    \n",
    "    return(wikidata_id)\n",
    "\n",
    "\n",
    "ids = []\n",
    "\n",
    "\n",
    "for index, row in obo_df_active_selection.iterrows():\n",
    "    print(index)\n",
    "    try:\n",
    "        ids.append(get_id(row[\"title\"].lower()))\n",
    "    except:\n",
    "        print(\"failed: \")\n",
    "        print(row[0])\n",
    "        ids.append(\"donotexist\")  \n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/lib/python3/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    }
   ],
   "source": [
    "obo_df_active_selection['wikidata_ids'] = ids"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "ename": "QueryBadFormed",
     "evalue": "QueryBadFormed: a bad request has been sent to the endpoint, probably the sparql query is bad formed. \n\nResponse:\nb'SPARQL-QUERY: queryStr=SELECT ?item ?itemLabel\\n            WHERE { \\n              ?item p:P31/ps:P31/wdt:P279* wd:Q324254.\\n              ?item rdfs:label ?itemLabel. \\n              FILTER(CONTAINS(LCASE(?itemLabel),xenopus anatomy ontology @en)). \\n            } limit 10\\n            \\njava.util.concurrent.ExecutionException: org.openrdf.query.MalformedQueryException: Lexical error at line 5, column 56.  Encountered: \" \" (32), after : \"xenopus\"\\n\\tat java.util.concurrent.FutureTask.report(FutureTask.java:122)\\n\\tat java.util.concurrent.FutureTask.get(FutureTask.java:206)\\n\\tat com.bigdata.rdf.sail.webapp.BigdataServlet.submitApiTask(BigdataServlet.java:292)\\n\\tat com.bigdata.rdf.sail.webapp.QueryServlet.doSparqlQuery(QueryServlet.java:678)\\n\\tat com.bigdata.rdf.sail.webapp.QueryServlet.doGet(QueryServlet.java:290)\\n\\tat com.bigdata.rdf.sail.webapp.RESTServlet.doGet(RESTServlet.java:240)\\n\\tat com.bigdata.rdf.sail.webapp.MultiTenancyServlet.doGet(MultiTenancyServlet.java:273)\\n\\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:687)\\n\\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:790)\\n\\tat org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:865)\\n\\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1655)\\n\\tat org.wikidata.query.rdf.blazegraph.throttling.ThrottlingFilter.doFilter(ThrottlingFilter.java:320)\\n\\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\\n\\tat org.wikidata.query.rdf.blazegraph.throttling.SystemOverloadFilter.doFilter(SystemOverloadFilter.java:82)\\n\\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\\n\\tat ch.qos.logback.classic.helpers.MDCInsertingServletFilter.doFilter(MDCInsertingServletFilter.java:49)\\n\\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\\n\\tat org.wikidata.query.rdf.blazegraph.filters.QueryEventSenderFilter.doFilter(QueryEventSenderFilter.java:93)\\n\\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\\n\\tat org.wikidata.query.rdf.blazegraph.filters.ClientIPFilter.doFilter(ClientIPFilter.java:43)\\n\\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\\n\\tat org.wikidata.query.rdf.blazegraph.filters.RealAgentFilter.doFilter(RealAgentFilter.java:33)\\n\\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1634)\\n\\tat org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533)\\n\\tat org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:146)\\n\\tat org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548)\\n\\tat org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)\\n\\tat org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:257)\\n\\tat org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)\\n\\tat org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)\\n\\tat org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1340)\\n\\tat org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)\\n\\tat org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)\\n\\tat org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564)\\n\\tat org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)\\n\\tat org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1242)\\n\\tat org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)\\n\\tat org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:220)\\n\\tat org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:126)\\n\\tat org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)\\n\\tat org.eclipse.jetty.server.Server.handle(Server.java:503)\\n\\tat org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:364)\\n\\tat org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:260)\\n\\tat org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:305)\\n\\tat org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)\\n\\tat org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118)\\n\\tat org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333)\\n\\tat org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310)\\n\\tat org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168)\\n\\tat org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126)\\n\\tat org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366)\\n\\tat org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:765)\\n\\tat org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:683)\\n\\tat java.lang.Thread.run(Thread.java:748)\\nCaused by: org.openrdf.query.MalformedQueryException: Lexical error at line 5, column 56.  Encountered: \" \" (32), after : \"xenopus\"\\n\\tat com.bigdata.rdf.sail.sparql.Bigdata2ASTSPARQLParser.parseQuery2(Bigdata2ASTSPARQLParser.java:404)\\n\\tat com.bigdata.rdf.sail.webapp.QueryServlet$SparqlQueryTask.call(QueryServlet.java:741)\\n\\tat com.bigdata.rdf.sail.webapp.QueryServlet$SparqlQueryTask.call(QueryServlet.java:695)\\n\\tat com.bigdata.rdf.task.ApiTaskForIndexManager.call(ApiTaskForIndexManager.java:68)\\n\\tat java.util.concurrent.FutureTask.run(FutureTask.java:266)\\n\\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\\n\\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\\n\\t... 1 more\\nCaused by: com.bigdata.rdf.sail.sparql.ast.TokenMgrError: Lexical error at line 5, column 56.  Encountered: \" \" (32), after : \"xenopus\"\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilderTokenManager.getNextToken(SyntaxTreeBuilderTokenManager.java:3994)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.jj_ntk(SyntaxTreeBuilder.java:9633)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.UnaryExpression(SyntaxTreeBuilder.java:4337)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.MultiplicativeExpression(SyntaxTreeBuilder.java:4283)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.AdditiveExpression(SyntaxTreeBuilder.java:4189)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.NumericExpression(SyntaxTreeBuilder.java:4185)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.RelationalExpression(SyntaxTreeBuilder.java:4013)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.ValueLogical(SyntaxTreeBuilder.java:4009)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.ConditionalAndExpression(SyntaxTreeBuilder.java:3973)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.ConditionalOrExpression(SyntaxTreeBuilder.java:3937)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.Expression(SyntaxTreeBuilder.java:3933)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.Contains(SyntaxTreeBuilder.java:6242)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.StringFunction(SyntaxTreeBuilder.java:5151)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.BuiltInCall(SyntaxTreeBuilder.java:5068)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.PrimaryExpression(SyntaxTreeBuilder.java:4556)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.UnaryExpression(SyntaxTreeBuilder.java:4455)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.MultiplicativeExpression(SyntaxTreeBuilder.java:4283)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.AdditiveExpression(SyntaxTreeBuilder.java:4189)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.NumericExpression(SyntaxTreeBuilder.java:4185)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.RelationalExpression(SyntaxTreeBuilder.java:4013)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.ValueLogical(SyntaxTreeBuilder.java:4009)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.ConditionalAndExpression(SyntaxTreeBuilder.java:3973)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.ConditionalOrExpression(SyntaxTreeBuilder.java:3937)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.Expression(SyntaxTreeBuilder.java:3933)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.BrackettedExpression(SyntaxTreeBuilder.java:5030)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.Constraint(SyntaxTreeBuilder.java:2569)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.Filter(SyntaxTreeBuilder.java:2558)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.FilterOrBind(SyntaxTreeBuilder.java:2241)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.BasicGraphPattern(SyntaxTreeBuilder.java:2107)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.GraphPattern(SyntaxTreeBuilder.java:2030)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.GroupGraphPattern(SyntaxTreeBuilder.java:1965)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.WhereClause(SyntaxTreeBuilder.java:1009)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.SelectQuery(SyntaxTreeBuilder.java:373)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.Query(SyntaxTreeBuilder.java:324)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.QueryContainer(SyntaxTreeBuilder.java:212)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.parseQuery(SyntaxTreeBuilder.java:32)\\n\\tat com.bigdata.rdf.sail.sparql.Bigdata2ASTSPARQLParser.parseQuery2(Bigdata2ASTSPARQLParser.java:336)\\n\\t... 7 more\\n'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mHTTPError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m/usr/local/lib/python3.6/dist-packages/SPARQLWrapper/Wrapper.py\u001b[0m in \u001b[0;36m_query\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    892\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 893\u001b[0;31m                 \u001b[0mresponse\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0murlopener\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    894\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mresponse\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturnFormat\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.6/urllib/request.py\u001b[0m in \u001b[0;36murlopen\u001b[0;34m(url, data, timeout, cafile, capath, cadefault, context)\u001b[0m\n\u001b[1;32m    222\u001b[0m         \u001b[0mopener\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_opener\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 223\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mopener\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    224\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.6/urllib/request.py\u001b[0m in \u001b[0;36mopen\u001b[0;34m(self, fullurl, data, timeout)\u001b[0m\n\u001b[1;32m    531\u001b[0m             \u001b[0mmeth\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mgetattr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mprocessor\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmeth_name\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 532\u001b[0;31m             \u001b[0mresponse\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmeth\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mreq\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mresponse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    533\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.6/urllib/request.py\u001b[0m in \u001b[0;36mhttp_response\u001b[0;34m(self, request, response)\u001b[0m\n\u001b[1;32m    641\u001b[0m             response = self.parent.error(\n\u001b[0;32m--> 642\u001b[0;31m                 'http', request, response, code, msg, hdrs)\n\u001b[0m\u001b[1;32m    643\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.6/urllib/request.py\u001b[0m in \u001b[0;36merror\u001b[0;34m(self, proto, *args)\u001b[0m\n\u001b[1;32m    569\u001b[0m             \u001b[0margs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mdict\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'default'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'http_error_default'\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0morig_args\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 570\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_call_chain\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    571\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.6/urllib/request.py\u001b[0m in \u001b[0;36m_call_chain\u001b[0;34m(self, chain, kind, meth_name, *args)\u001b[0m\n\u001b[1;32m    503\u001b[0m             \u001b[0mfunc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mgetattr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhandler\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmeth_name\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 504\u001b[0;31m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    505\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mresult\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.6/urllib/request.py\u001b[0m in \u001b[0;36mhttp_error_default\u001b[0;34m(self, req, fp, code, msg, hdrs)\u001b[0m\n\u001b[1;32m    649\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mhttp_error_default\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mreq\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfp\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmsg\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhdrs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 650\u001b[0;31m         \u001b[0;32mraise\u001b[0m \u001b[0mHTTPError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mreq\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfull_url\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmsg\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhdrs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfp\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    651\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mHTTPError\u001b[0m: HTTP Error 400: Bad Request",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mQueryBadFormed\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-14-6c22b92d810c>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mget_id\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'Xenopus Anatomy Ontology'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m<ipython-input-13-87806032f415>\u001b[0m in \u001b[0;36mget_id\u001b[0;34m(name)\u001b[0m\n\u001b[1;32m     21\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     22\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 23\u001b[0;31m     \u001b[0mresults\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mget_results\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mendpoint_url\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mquery\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     24\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     25\u001b[0m     \u001b[0mwikidata_id\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mresults\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'results'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'bindings'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'item'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'value'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'/'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m4\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-13-87806032f415>\u001b[0m in \u001b[0;36mget_results\u001b[0;34m(endpoint_url, query)\u001b[0m\n\u001b[1;32m     18\u001b[0m         \u001b[0msparql\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msetQuery\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mquery\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     19\u001b[0m         \u001b[0msparql\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msetReturnFormat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mJSON\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 20\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0msparql\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mquery\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mconvert\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     21\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     22\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/local/lib/python3.6/dist-packages/SPARQLWrapper/Wrapper.py\u001b[0m in \u001b[0;36mquery\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    925\u001b[0m             \u001b[0;34m@\u001b[0m\u001b[0mrtype\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mL\u001b[0m\u001b[0;34m{\u001b[0m\u001b[0mQueryResult\u001b[0m\u001b[0;34m}\u001b[0m \u001b[0minstance\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    926\u001b[0m         \"\"\"\n\u001b[0;32m--> 927\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mQueryResult\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_query\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    928\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    929\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mqueryAndConvert\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/local/lib/python3.6/dist-packages/SPARQLWrapper/Wrapper.py\u001b[0m in \u001b[0;36m_query\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    895\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0murllib\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0merror\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mHTTPError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    896\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcode\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;36m400\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 897\u001b[0;31m                 \u001b[0;32mraise\u001b[0m \u001b[0mQueryBadFormed\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    898\u001b[0m             \u001b[0;32melif\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcode\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;36m404\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    899\u001b[0m                 \u001b[0;32mraise\u001b[0m \u001b[0mEndPointNotFound\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mQueryBadFormed\u001b[0m: QueryBadFormed: a bad request has been sent to the endpoint, probably the sparql query is bad formed. \n\nResponse:\nb'SPARQL-QUERY: queryStr=SELECT ?item ?itemLabel\\n            WHERE { \\n              ?item p:P31/ps:P31/wdt:P279* wd:Q324254.\\n              ?item rdfs:label ?itemLabel. \\n              FILTER(CONTAINS(LCASE(?itemLabel),xenopus anatomy ontology @en)). \\n            } limit 10\\n            \\njava.util.concurrent.ExecutionException: org.openrdf.query.MalformedQueryException: Lexical error at line 5, column 56.  Encountered: \" \" (32), after : \"xenopus\"\\n\\tat java.util.concurrent.FutureTask.report(FutureTask.java:122)\\n\\tat java.util.concurrent.FutureTask.get(FutureTask.java:206)\\n\\tat com.bigdata.rdf.sail.webapp.BigdataServlet.submitApiTask(BigdataServlet.java:292)\\n\\tat com.bigdata.rdf.sail.webapp.QueryServlet.doSparqlQuery(QueryServlet.java:678)\\n\\tat com.bigdata.rdf.sail.webapp.QueryServlet.doGet(QueryServlet.java:290)\\n\\tat com.bigdata.rdf.sail.webapp.RESTServlet.doGet(RESTServlet.java:240)\\n\\tat com.bigdata.rdf.sail.webapp.MultiTenancyServlet.doGet(MultiTenancyServlet.java:273)\\n\\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:687)\\n\\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:790)\\n\\tat org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:865)\\n\\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1655)\\n\\tat org.wikidata.query.rdf.blazegraph.throttling.ThrottlingFilter.doFilter(ThrottlingFilter.java:320)\\n\\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\\n\\tat org.wikidata.query.rdf.blazegraph.throttling.SystemOverloadFilter.doFilter(SystemOverloadFilter.java:82)\\n\\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\\n\\tat ch.qos.logback.classic.helpers.MDCInsertingServletFilter.doFilter(MDCInsertingServletFilter.java:49)\\n\\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\\n\\tat org.wikidata.query.rdf.blazegraph.filters.QueryEventSenderFilter.doFilter(QueryEventSenderFilter.java:93)\\n\\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\\n\\tat org.wikidata.query.rdf.blazegraph.filters.ClientIPFilter.doFilter(ClientIPFilter.java:43)\\n\\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\\n\\tat org.wikidata.query.rdf.blazegraph.filters.RealAgentFilter.doFilter(RealAgentFilter.java:33)\\n\\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1634)\\n\\tat org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533)\\n\\tat org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:146)\\n\\tat org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548)\\n\\tat org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)\\n\\tat org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:257)\\n\\tat org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)\\n\\tat org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)\\n\\tat org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1340)\\n\\tat org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)\\n\\tat org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)\\n\\tat org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564)\\n\\tat org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)\\n\\tat org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1242)\\n\\tat org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)\\n\\tat org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:220)\\n\\tat org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:126)\\n\\tat org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)\\n\\tat org.eclipse.jetty.server.Server.handle(Server.java:503)\\n\\tat org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:364)\\n\\tat org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:260)\\n\\tat org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:305)\\n\\tat org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)\\n\\tat org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118)\\n\\tat org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333)\\n\\tat org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310)\\n\\tat org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168)\\n\\tat org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126)\\n\\tat org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366)\\n\\tat org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:765)\\n\\tat org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:683)\\n\\tat java.lang.Thread.run(Thread.java:748)\\nCaused by: org.openrdf.query.MalformedQueryException: Lexical error at line 5, column 56.  Encountered: \" \" (32), after : \"xenopus\"\\n\\tat com.bigdata.rdf.sail.sparql.Bigdata2ASTSPARQLParser.parseQuery2(Bigdata2ASTSPARQLParser.java:404)\\n\\tat com.bigdata.rdf.sail.webapp.QueryServlet$SparqlQueryTask.call(QueryServlet.java:741)\\n\\tat com.bigdata.rdf.sail.webapp.QueryServlet$SparqlQueryTask.call(QueryServlet.java:695)\\n\\tat com.bigdata.rdf.task.ApiTaskForIndexManager.call(ApiTaskForIndexManager.java:68)\\n\\tat java.util.concurrent.FutureTask.run(FutureTask.java:266)\\n\\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\\n\\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\\n\\t... 1 more\\nCaused by: com.bigdata.rdf.sail.sparql.ast.TokenMgrError: Lexical error at line 5, column 56.  Encountered: \" \" (32), after : \"xenopus\"\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilderTokenManager.getNextToken(SyntaxTreeBuilderTokenManager.java:3994)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.jj_ntk(SyntaxTreeBuilder.java:9633)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.UnaryExpression(SyntaxTreeBuilder.java:4337)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.MultiplicativeExpression(SyntaxTreeBuilder.java:4283)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.AdditiveExpression(SyntaxTreeBuilder.java:4189)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.NumericExpression(SyntaxTreeBuilder.java:4185)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.RelationalExpression(SyntaxTreeBuilder.java:4013)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.ValueLogical(SyntaxTreeBuilder.java:4009)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.ConditionalAndExpression(SyntaxTreeBuilder.java:3973)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.ConditionalOrExpression(SyntaxTreeBuilder.java:3937)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.Expression(SyntaxTreeBuilder.java:3933)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.Contains(SyntaxTreeBuilder.java:6242)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.StringFunction(SyntaxTreeBuilder.java:5151)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.BuiltInCall(SyntaxTreeBuilder.java:5068)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.PrimaryExpression(SyntaxTreeBuilder.java:4556)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.UnaryExpression(SyntaxTreeBuilder.java:4455)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.MultiplicativeExpression(SyntaxTreeBuilder.java:4283)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.AdditiveExpression(SyntaxTreeBuilder.java:4189)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.NumericExpression(SyntaxTreeBuilder.java:4185)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.RelationalExpression(SyntaxTreeBuilder.java:4013)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.ValueLogical(SyntaxTreeBuilder.java:4009)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.ConditionalAndExpression(SyntaxTreeBuilder.java:3973)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.ConditionalOrExpression(SyntaxTreeBuilder.java:3937)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.Expression(SyntaxTreeBuilder.java:3933)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.BrackettedExpression(SyntaxTreeBuilder.java:5030)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.Constraint(SyntaxTreeBuilder.java:2569)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.Filter(SyntaxTreeBuilder.java:2558)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.FilterOrBind(SyntaxTreeBuilder.java:2241)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.BasicGraphPattern(SyntaxTreeBuilder.java:2107)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.GraphPattern(SyntaxTreeBuilder.java:2030)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.GroupGraphPattern(SyntaxTreeBuilder.java:1965)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.WhereClause(SyntaxTreeBuilder.java:1009)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.SelectQuery(SyntaxTreeBuilder.java:373)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.Query(SyntaxTreeBuilder.java:324)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.QueryContainer(SyntaxTreeBuilder.java:212)\\n\\tat com.bigdata.rdf.sail.sparql.ast.SyntaxTreeBuilder.parseQuery(SyntaxTreeBuilder.java:32)\\n\\tat com.bigdata.rdf.sail.sparql.Bigdata2ASTSPARQLParser.parseQuery2(Bigdata2ASTSPARQLParser.java:336)\\n\\t... 7 more\\n'"
     ]
    }
   ],
   "source": [
    "get_id('Xenopus Anatomy Ontology')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Q4866972|P31|Q81314568\n",
      "Q4866972|P856|http://ifomis.org/bfo/\n",
      "Q4866972|P361|Q7072326\n",
      "\n",
      "Q902623|P31|Q81314568\n",
      "Q902623|P856|http://www.ebi.ac.uk/chebi\n",
      "Q902623|P361|Q7072326\n",
      "\n",
      "Q58587067|P31|Q81314568\n",
      "Q58587067|P856|http://www.disease-ontology.org\n",
      "Q58587067|P361|Q7072326\n",
      "\n",
      "Q135085|P31|Q81314568\n",
      "Q135085|P856|http://geneontology.org/\n",
      "Q135085|P361|Q7072326\n",
      "\n",
      "Q7095051|P31|Q81314568\n",
      "Q7095051|P856|http://obi-ontology.org\n",
      "Q7095051|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Phenotype And Trait Ontology\"\n",
      "LAST|Den|\"An ontology of phenotypic qualities (properties, attributes or characteristics)\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/pato-ontology/pato/\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q7201529|P31|Q81314568\n",
      "Q7201529|P856|http://browser.planteome.org/amigo\n",
      "Q7201529|P361|Q7072326\n",
      "\n",
      "Q37556803|P31|Q81314568\n",
      "Q37556803|P856|http://proconsortium.org\n",
      "Q37556803|P361|Q7072326\n",
      "\n",
      "Q55118650|P31|Q81314568\n",
      "Q55118650|P856|http://www.xenbase.org/anatomy/xao.do?method=display\n",
      "Q55118650|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Zebrafish anatomy and development ontology\"\n",
      "LAST|Den|\"A structured controlled vocabulary of the anatomy and development of the Zebrafish\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://wiki.zfin.org/display/general/Anatomy+Atlases+and+Resources\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Anatomical Entity Ontology\"\n",
      "LAST|Den|\"AEO is an ontology of anatomical structures that expands CARO, the Common Anatomy Reference Ontology\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/obophenotype/human-developmental-anatomy-ontology/\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Agronomy Ontology\"\n",
      "LAST|Den|\"Ontology of agronomic practices, agronomic techniques, and agronomic variables used in agronomic experiments\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/AgriculturalSemantics/agro\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Ascomycete phenotype ontology\"\n",
      "LAST|Den|\"A structured controlled vocabulary for the phenotypes of Ascomycete fungi\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://www.yeastgenome.org/\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q55118240|P31|Q81314568\n",
      "Q55118240|P856|https://github.com/ApolloDev/apollo-sv\n",
      "Q55118240|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Antibiotic Resistance Ontology\"\n",
      "LAST|Den|\"Antibiotic resistance genes and mutations\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/arpcard/aro\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Beta Cell Genomics Ontology\"\n",
      "LAST|Den|\"An application ontology built for beta cell genomics studies.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/obi-bcgo/bcgo\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q55118250|P31|Q81314568\n",
      "Q55118250|P856|https://github.com/tucotuco/bco\n",
      "Q55118250|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Biological Spatial Ontology\"\n",
      "LAST|Den|\"An ontology for respresenting spatial concepts, anatomical axes, gradients, regions, planes, sides, and surfaces\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/obophenotype/biological-spatial-ontology\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"BRENDA tissue / enzyme source\"\n",
      "LAST|Den|\"A structured controlled vocabulary for the source of an enzyme comprising tissues, cell lines, cell types and cell cultures.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://www.brenda-enzymes.org\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q55118319|P31|Q81314568\n",
      "Q55118319|P856|https://github.com/obophenotype/caro/\n",
      "Q55118319|P361|Q7072326\n",
      "\n",
      "Q55118328|P31|Q81314568\n",
      "Q55118328|P856|https://github.com/evoinfo/cdao\n",
      "Q55118328|P361|Q7072326\n",
      "\n",
      "Q55118298|P31|Q81314568\n",
      "Q55118298|P856|https://github.com/semanticchemistry/semanticchemistry\n",
      "Q55118298|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"CHEBI Integrated Role Ontology\"\n",
      "LAST|Den|\"CHEBI provides a distinct role hierarchy. Chemicals in the structural hierarchy are connected via a 'has role' relation. CHIRO provides links from these roles to useful other classes in other ontologies. This will allow direct connection between chemical structures (small molecules, drugs) and what they do. This could be formalized using 'capable of', in the same way Uberon and the Cell Ontology link structures to processes.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/obophenotype/chiro\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q55118301|P31|Q81314568\n",
      "Q55118301|P856|https://github.com/rsc-ontologies/rsc-cmo\n",
      "Q55118301|P361|Q7072326\n",
      "\n",
      "Q55118285|P31|Q81314568\n",
      "Q55118285|P856|https://github.com/obophenotype/cell-ontology\n",
      "Q55118285|P361|Q7072326\n",
      "\n",
      "Q21039006|P31|Q81314568\n",
      "Q21039006|P856|http://www.clo-ontology.org\n",
      "Q21039006|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Clinical measurement ontology\"\n",
      "LAST|Den|\"Morphological and physiological measurement records generated from clinical and model organism research and health programs.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://rgd.mcw.edu/rgdweb/ontology/search.html\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q70987840|P31|Q81314568\n",
      "Q70987840|P856|https://github.com/data2health/contributor-role-ontology\n",
      "Q70987840|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Ctenophore Ontology\"\n",
      "LAST|Den|\"An anatomical and developmental ontology for ctenophores (Comb Jellies)\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/obophenotype/ctenophore-ontology\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q55118269|P31|Q81314568\n",
      "Q55118269|P856|https://github.com/OpenLHS/CVDO\n",
      "Q55118269|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Dictyostelium discoideum anatomy\"\n",
      "LAST|Den|\"A structured controlled vocabulary of the anatomy of the slime-mold Dictyostelium discoideum\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://dictybase.org/\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Dictyostelium discoideum phenotype ontology\"\n",
      "LAST|Den|\"A structured controlled vocabulary of phenotypes of the slime-mould <i>Dictyostelium discoideum</i>.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://dictybase.org/\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Drug-drug Interaction and Drug-drug Interaction Evidence Ontology\"\n",
      "LAST|Den|\"The Potential Drug-drug Interaction and Potential Drug-drug Interaction Evidence Ontology\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/DIDEO/DIDEO\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"The Drug-Drug Interactions Ontology\"\n",
      "LAST|Den|\"A formal represention for drug-drug interactions knowledge.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://labda.inf.uc3m.es/doku.php?id=es:labda_dinto\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Drosophila Phenotype Ontology\"\n",
      "LAST|Den|\"An ontology of commonly encountered and/or high level Drosophila phenotypes.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://purl.obolibrary.org/obo/fbcv\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"The Drug Ontology\"\n",
      "LAST|Den|\"An ontology to support comparative effectiveness researchers studying claims data.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://bitbucket.org/uamsdbmi/dron.git\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"The Data Use Ontology\"\n",
      "LAST|Den|\"DUO is an ontology which represent data use conditions.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/EBISPOT/DUO\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Evidence ontology\"\n",
      "LAST|Den|\"An ontology for experimental and other evidence statements.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/evidenceontology/evidenceontology/\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"An ontology of core ecological entities\"\n",
      "LAST|Den|\"An ontology to provide core semantics for ecological entities.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/EcologicalSemantics/ecocore\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Environmental conditions, treatments and exposures ontology\"\n",
      "LAST|Den|\"ECTO describes exposures to experimental treatments of plants and model organisms (e.g. exposures to modification of diet, lighting levels, temperature); exposures of humans or any other organisms to stressors through a variety of routes, for purposes of public health, environmental monitoring etc, stimuli, natural and experimental, any kind of environmental condition or change in condition that can be experienced by an organism or population of organisms on earth. The scope is very general and can include for example plant treatment regimens, as well as human clinical exposures (although these may better be handled by a more specialized ontology).\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/EnvironmentOntology/environmental-exposure-ontology\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Mouse gross anatomy and development, timed\"\n",
      "LAST|Den|\"Abstract (i.e. non-stage specific) stage-specific anatomical structures of the mouse.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://emouseatlas.org\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q31110555|P31|Q81314568\n",
      "Q31110555|P856|http://environmentontology.org/\n",
      "Q31110555|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"eagle-i resource ontology\"\n",
      "LAST|Den|\"An ontology of research resources such as instruments. protocols, reagents, animal models and biospecimens.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://open.med.harvard.edu/wiki/display/eaglei/Ontology\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"VEuPathDB ontology\"\n",
      "LAST|Den|\"An ontology is developed to support Eukaryotic Pathogen, Host & Vector Genomics Resource (VEuPathDB; https://veupathdb.org).\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/VEuPathDB-ontology/VEuPathDB-ontology\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Exposure ontology\"\n",
      "LAST|Den|\"Vocabularies for describing exposure data to inform understanding of environmental health.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://ctdbase.org/help/exposureHelp.jsp\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Fungal gross anatomy\"\n",
      "LAST|Den|\"A structured controlled vocabulary for the anatomy of fungi.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/obophenotype/fungal-anatomy-ontology/\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Biological Imaging Methods Ontology\"\n",
      "LAST|Den|\"A structured controlled vocabulary of sample preparation, visualization and imaging methods used in biomedical research.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://cellimagelibrary.org/\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Drosophila gross anatomy\"\n",
      "LAST|Den|\"An ontology representing the gross anatomy of Drosophila melanogaster.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://purl.obolibrary.org/obo/fbbt\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"FlyBase Controlled Vocabulary\"\n",
      "LAST|Den|\"A structured controlled vocabulary used for various aspects of annotation by FlyBase.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://purl.obolibrary.org/obo/fbcv\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q36235170|P31|Q81314568\n",
      "Q36235170|P856|http://purl.obolibrary.org/obo/fbdv\n",
      "Q36235170|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Flora Phenotype Ontology\"\n",
      "LAST|Den|\"Traits and phenotypes of flowering plants occurring in digitized Floras\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/flora-phenotype-ontology/flopoontology\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Foundational Model of Anatomy Ontology (subset)\"\n",
      "LAST|Den|\"This is currently a slimmed down version of FMA\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://si.washington.edu/projects/fma\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q55118395|P31|Q81314568\n",
      "Q55118395|P856|http://foodontology.github.io/foodon/\n",
      "Q55118395|P361|Q7072326\n",
      "\n",
      "Q55118393|P31|Q81314568\n",
      "Q55118393|P856|https://github.com/pombase/fypo\n",
      "Q55118393|P361|Q7072326\n",
      "\n",
      "Q1006160|P31|Q81314568\n",
      "Q1006160|P856|http://environmentontology.github.io/gaz/\n",
      "Q1006160|P361|Q7072326\n",
      "\n",
      "Q55118402|P31|Q81314568\n",
      "Q55118402|P856|http://genepio.org/\n",
      "Q55118402|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Genotype Ontology\"\n",
      "LAST|Den|\"An integrated ontology for representing the genetic variations described in genotypes, and their causal relationships to phenotype and diseases.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/monarch-initiative/GENO-ontology/\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Geographical Entity Ontology\"\n",
      "LAST|Den|\"An ontology of geographical entities\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/ufbmi/geographical-entity-ontology/wiki\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Glycan Naming Ontology\"\n",
      "LAST|Den|\"An ontology for glycans based on GlyTouCan, but organized by subsumption.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/glygen-glycan-data/GNOme\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Human Ancestry Ontology\"\n",
      "LAST|Den|\"The Human Ancestry Ontology (HANCESTRO) provides a systematic description of the ancestry concepts used in the NHGRI-EBI Catalog of published genome-wide association studies.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/EBISPOT/ancestro\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Hymenoptera Anatomy Ontology\"\n",
      "LAST|Den|\"A structured controlled vocabulary of the anatomy of the Hymenoptera (bees, wasps, and ants)\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://hymao.org\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Homology Ontology\"\n",
      "LAST|Den|\"This ontology represents concepts related to homology, as well as other concepts used to describe similarity and non-homology.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/BgeeDB/homology-ontology\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Human Developmental Stages\"\n",
      "LAST|Den|\"Life cycle stages for Human\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/obophenotype/developmental-stage-ontologies/wiki/HsapDv\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Hypertension Ontology\"\n",
      "LAST|Den|\"An ontology for representing clinical data about hypertension, intended to support classification of patients according to various diagnostic guidelines\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/aellenhicks/htn_owl\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q55118415|P31|Q81314568\n",
      "Q55118415|P856|https://github.com/information-artifact-ontology/IAO/\n",
      "Q55118415|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Integrative and Conjugative Element Ontology\"\n",
      "LAST|Den|\"ICEO is an integrated biological ontology for the description of bacterial integrative and conjugative elements (ICEs).\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/ontoice/ICEO\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q55118417|P31|Q81314568\n",
      "Q55118417|P856|https://github.com/ICO-ontology/ICO\n",
      "Q55118417|P361|Q7072326\n",
      "\n",
      "Q54302665|P31|Q81314568\n",
      "Q54302665|P856|http://www.bioontology.org/wiki/index.php/Infectious_Disease_Ontology\n",
      "Q54302665|P361|Q7072326\n",
      "\n",
      "Q55118437|P31|Q81314568\n",
      "Q55118437|P856|https://www.vectorbase.org/ontology-browser\n",
      "Q55118437|P361|Q7072326\n",
      "\n",
      "Q55118418|P31|Q81314568\n",
      "Q55118418|P856|https://github.com/INO-ontology/ino\n",
      "Q55118418|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"clinical LABoratory Ontology\"\n",
      "LAST|Den|\"LABO is an ontology of informational entities formalizing clinical laboratory tests prescriptions and reporting documents.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/OpenLHS/LABO\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Mouse adult gross anatomy\"\n",
      "LAST|Den|\"A structured controlled vocabulary of the adult anatomy of the mouse (Mus).\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/obophenotype/mouse-anatomy-ontology\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Microbial Conditions Ontology\"\n",
      "LAST|Den|\"Microbial Conditions Ontology is an ontology...\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/microbial-conditions-ontology/microbial-conditions-ontology\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q55118443|P31|Q81314568\n",
      "Q55118443|P856|https://github.com/jannahastings/mental-functioning-ontology\n",
      "Q55118443|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Mammalian Feeding Muscle Ontology\"\n",
      "LAST|Den|\"The Mammalian Feeding Muscle Ontology is an antomy ontology for the muscles of the head and neck that participate in feeding, swallowing, and other oral-pharyngeal behaviors.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/rdruzinsky/feedontology\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q55118370|P31|Q81314568\n",
      "Q55118370|P856|https://github.com/jannahastings/emotion-ontology\n",
      "Q55118370|P361|Q7072326\n",
      "\n",
      "Q55118439|P31|Q81314568\n",
      "Q55118439|P856|https://github.com/jannahastings/mental-functioning-ontology\n",
      "Q55118439|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Molecular Interactions Controlled Vocabulary\"\n",
      "LAST|Den|\"A structured controlled vocabulary for the annotation of experiments concerned with protein-protein interactions.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/HUPO-PSI/psi-mi-CV\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"MIAPA Ontology\"\n",
      "LAST|Den|\"An application ontology to formalize annotation of phylogenetic data.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://www.evoio.org/wiki/MIAPA\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Ontology of Prokaryotic Phenotypic and Metabolic Characters\"\n",
      "LAST|Den|\"An ontology of prokaryotic phenotypic and metabolic characters\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/carrineblank/MicrO\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"microRNA Ontology\"\n",
      "LAST|Den|\"An application ontology for use with miRNA databases.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://code.google.com/p/mirna-ontology/\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Measurement method ontology\"\n",
      "LAST|Den|\"A representation of the variety of methods used to make clinical and phenotype measurements. \"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://rgd.mcw.edu/rgdweb/ontology/view.html?acc_id=MMO:0000000\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Mouse Developmental Stages\"\n",
      "LAST|Den|\"Life cycle stages for Mus Musculus\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/obophenotype/developmental-stage-ontologies/wiki/MmusDv\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q66735349|P31|Q81314568\n",
      "Q66735349|P856|http://www.psidev.info/MOD\n",
      "Q66735349|P361|Q7072326\n",
      "\n",
      "Q27468140|P31|Q81314568\n",
      "Q27468140|P856|https://monarch-initiative.github.io/mondo\n",
      "Q27468140|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Molecular Process Ontology\"\n",
      "LAST|Den|\"Processes at the molecular level\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/rsc-ontologies/rxno\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Mammalian Phenotype Ontology\"\n",
      "LAST|Den|\"Standard terms for annotating mammalian phenotypic data.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://www.informatics.jax.org/searches/MP_form.shtml\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q55118458|P31|Q81314568\n",
      "Q55118458|P856|http://www.pathbase.net\n",
      "Q55118458|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Minimum PDDI Information Ontology\"\n",
      "LAST|Den|\"An ontology of minimum information regarding potential drug-drug interaction information.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/MPIO-Developers/MPIO\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"MHC Restriction Ontology\"\n",
      "LAST|Den|\"An ontology for Major Histocompatibility Complex (MHC) restriction in experiments\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/IEDB/MRO\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Mass spectrometry ontology\"\n",
      "LAST|Den|\"A structured controlled vocabulary for the annotation of experiments concerned with proteomics mass spectrometry.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://www.psidev.info/groups/controlled-vocabularies\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Neuro Behavior Ontology\"\n",
      "LAST|Den|\"An ontology of human and animal behaviours and behavioural phenotypes\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/obo-behavior/behavior-ontology/\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"NCBI organismal classification\"\n",
      "LAST|Den|\"An ontology representation of the NCBI organismal taxonomy\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/obophenotype/ncbitaxon\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"NCI Thesaurus OBO Edition\"\n",
      "LAST|Den|\"NCI Thesaurus (NCIt)is a reference terminology that includes broad coverage of the cancer domain, including cancer related diseases, findings and abnormalities. The NCIt OBO Edition aims to increase integration of the NCIt with OBO Library ontologies. NCIt OBO Edition releases should be considered experimental.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/NCI-Thesaurus/thesaurus-obo-edition\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q55118485|P31|Q81314568\n",
      "Q55118485|P856|http://omnisearch.soc.southalabama.edu/w/index.php/Ontology\n",
      "Q55118485|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"NOMEN - A nomenclatural ontology for biological names\"\n",
      "LAST|Den|\"NOMEN is a nomenclatural ontology for biological names (not concepts).  It encodes the goverened rules of nomenclature.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/SpeciesFileGroup/nomen\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Ontology of Adverse Events\"\n",
      "LAST|Den|\"A biomedical ontology in the domain of adverse events\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/OAE-ontology/OAE/\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Ontology of Arthropod Circulatory Systems\"\n",
      "LAST|Den|\"OArCS is an ontology describing the Arthropod ciruclatory system.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/aszool/oarcs\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Ontology of Biological Attributes\"\n",
      "LAST|Den|\"A collection of biological attributes (traits) covering all kingdoms of life.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/obophenotype/bio-attribute-ontology\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q55118544|P31|Q81314568\n",
      "Q55118544|P856|https://github.com/obcs/obcs\n",
      "Q55118544|P361|Q7072326\n",
      "\n",
      "Q55118499|P31|Q81314568\n",
      "Q55118499|P856|https://github.com/biobanking/biobanking\n",
      "Q55118499|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"The Ontology of Genes and Genomes\"\n",
      "LAST|Den|\"A formal ontology of genes and genomes of biological organisms.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://bitbucket.org/hegroup/ogg\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q55118512|P31|Q81314568\n",
      "Q55118512|P856|https://github.com/OGMS/ogms\n",
      "Q55118512|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Ontology of Genetic Susceptibility Factor\"\n",
      "LAST|Den|\"An application ontology to represent genetic susceptibility to a specific disease, adverse event, or a pathological process.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|none\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"The Oral Health and Disease Ontology\"\n",
      "LAST|Den|\"The Oral Health and Disease Ontology was created, initially, to represent the content of dental practice health records.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://purl.obolibrary.org/obo/ohd/home\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Ontology of Host-Microbiome Interactions\"\n",
      "LAST|Den|\"The Ontology of Host-Microbiome Interactions aims to ontologically represent and standardize various entities and relations related to microbiomes, microbiome host organisms (e.g., human and mouse), and the interactions between the hosts and microbiomes at different conditions.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/ohmi-ontology/ohmi\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Ontology of Host Pathogen Interactions\"\n",
      "LAST|Den|\"OHPI is a community-driven ontology of host-pathogen interactions (OHPI) and represents the virulence factors (VFs) and how the mutants of VFs in the Victors database become less virulence inside a host organism or host cells. It is developed to represent manually curated HPI knowledge available in the PHIDIAS resource.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/OHPI/ohpi\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Medaka Developmental Stages\"\n",
      "LAST|Den|\"Life cycle stages for Medaka\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/obophenotype/developmental-stage-ontologies/wiki/OlatDv\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Ontologized MIABIS\"\n",
      "LAST|Den|\"An ontological version of MIABIS (Minimum Information About BIobank data Sharing)\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/OMIABIS/omiabis-dev\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Ontology for MIRNA Target\"\n",
      "LAST|Den|\"Ontology to establish data exchange standards and common data elements in the microRNA (miR) domain\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://omit.cis.usouthal.edu/\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Ontology of Microbial Phenotypes\"\n",
      "LAST|Den|\"An ontology of phenotypes covering microbes\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://microbialphenotypes.org\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q55118551|P31|Q81314568\n",
      "Q55118551|P856|https://github.com/ufbmi/OMRSE/wiki/OMRSE-Overview\n",
      "Q55118551|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Obstetric and Neonatal Ontology\"\n",
      "LAST|Den|\"The Obstetric and Neonatal Ontology is a structured controlled vocabulary to provide a representation of the data from electronic health records (EHRs) involved in the care of the pregnant woman, and of her baby.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|ontoneo.com\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Ontology of Organizational Structures of Trauma centers and Trauma systems\"\n",
      "LAST|Den|\"An ontology built for representating the organizational components of trauma centers and trauma systems.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/OOSTT/OOSTT\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q55118540|P31|Q81314568\n",
      "Q55118540|P856|https://github.com/OPL-ontology/OPL\n",
      "Q55118540|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Ontology of Precision Medicine and Investigation\"\n",
      "LAST|Den|\"The Ontology of Precision Medicine and Investigation (OPMI) aims to ontologically represent and standardize various entities and relations associated with precision medicine and related investigations at different conditions.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/OPMI/opmi\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Ontology of RNA Sequencing\"\n",
      "LAST|Den|\"An application ontology designed to annotate next-generation sequencing experiments performed on RNA.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://kim.bio.upenn.edu/software/ornaseq.shtml\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q55118555|P31|Q81314568\n",
      "Q55118555|P856|http://www.violinet.org/ovae/\n",
      "Q55118555|P361|Q7072326\n",
      "\n",
      "Q55118575|P31|Q81314568\n",
      "Q55118575|P856|https://github.com/PopulationAndCommunityOntology/pco\n",
      "Q55118575|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"The Prescription of Drugs Ontology\"\n",
      "LAST|Den|\"An ontology to describe entities related to prescription of drugs\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/OpenLHS/PDRO\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Platynereis Developmental Stages\"\n",
      "LAST|Den|\"Life cycle stages for Platynereis dumerilii\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/obophenotype/developmental-stage-ontologies/wiki/PdumDv\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Plant Experimental Conditions Ontology\"\n",
      "LAST|Den|\"A structured, controlled vocabulary which describes the treatments, growing conditions, and/or study types used in plant biology experiments.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://planteome.org/\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Pathogen Host Interaction Phenotype Ontology\"\n",
      "LAST|Den|\"PHIPO is a formal ontology of species-neutral phenotypes observed in pathogen-host interactions.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/PHI-base/phipo\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"planaria-ontology\"\n",
      "LAST|Den|\"planaria-ontology is an ontology for planarian anatomy and developmental stages of S.med\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/obophenotype/planaria-ontology\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Planarian Phenotype Ontology\"\n",
      "LAST|Den|\"Planarian Phenotype Ontology is an ontology of phenotypes observed in the planarian Schmidtea mediterranea.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/obophenotype/planarian-phenotype-ontology\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q55118578|P31|Q81314568\n",
      "Q55118578|P856|https://github.com/obophenotype/porifera-ontology\n",
      "Q55118578|P361|Q7072326\n",
      "\n",
      "Q65057180|P31|Q81314568\n",
      "Q65057180|P856|https://github.com/PlantPhenoOntology/PPO\n",
      "Q65057180|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Performance Summary Display Ontology\"\n",
      "LAST|Den|\"Ontology to reproducibly study visualizations of clinical performance\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/Display-Lab/psdo\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q28864280|P31|Q81314568\n",
      "Q28864280|P856|http://rgd.mcw.edu/rgdweb/ontology/search.html\n",
      "Q28864280|P361|Q7072326\n",
      "\n",
      "Q55118589|P31|Q81314568\n",
      "Q55118589|P856|https://github.com/bgsu-rna/rnao\n",
      "Q55118589|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Relation Ontology\"\n",
      "LAST|Den|\"Relationship types shared across multiple ontologies\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/oborel/obo-relations/\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q30699801|P31|Q81314568\n",
      "Q30699801|P856|http://rgd.mcw.edu/rgdweb/search/strains.html\n",
      "Q30699801|P361|Q7072326\n",
      "\n",
      "Q20991891|P31|Q81314568\n",
      "Q20991891|P856|https://github.com/rsc-ontologies/rxno\n",
      "Q20991891|P361|Q7072326\n",
      "\n",
      "Q55118599|P31|Q81314568\n",
      "Q55118599|P856|https://github.com/monarch-initiative/SEPIO-ontology\n",
      "Q55118599|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Sequence types and features ontology\"\n",
      "LAST|Den|\"A structured controlled vocabulary for sequence annotation, for the exchange of annotation data and for the description of sequence objects in databases.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/The-Sequence-Ontology/SO-Ontologies\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q42404539|P31|Q81314568\n",
      "Q42404539|P856|http://research.amnh.org/atol/files/\n",
      "Q42404539|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"The Statistical Methods Ontology\"\n",
      "LAST|Den|\"STATO is a general-purpose STATistics Ontology. Its aim is to provide coverage for processes such as statistical tests, their conditions of application, and information needed or resulting from statistical methods, such as probability distributions, variables, spread and variation metrics. STATO also covers aspects of experimental design and description of plots and graphical representations commonly used to provide visual cues of data distribution or layout and to assist review of the results.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://stato-ontology.org/\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q55118620|P31|Q81314568\n",
      "Q55118620|P856|https://github.com/allysonlister/swo\n",
      "Q55118620|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Symptom Ontology\"\n",
      "LAST|Den|\"An ontology of disease symptoms, with symptoms encompasing perceived changes in function, sensations or appearance reported by a patient indicative of a disease.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://symptomontologywiki.igs.umaryland.edu/mediawiki/index.php/Main_Page\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Tick Anatomy Ontology\"\n",
      "LAST|Den|\"The anatomy of the Tick, <i>Families: Ixodidae, Argassidae</i>\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://www.vectorbase.org/ontology-browser\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Taxonomic rank vocabulary\"\n",
      "LAST|Den|\"A vocabulary of taxonomic ranks (species, family, phylum, etc)\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/phenoscape/taxrank\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Mosquito gross anatomy ontology\"\n",
      "LAST|Den|\"A structured controlled vocabulary of the anatomy of mosquitoes.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://anobase.vectorbase.org\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Plant Trait Ontology\"\n",
      "LAST|Den|\"A controlled vocabulary of describe phenotypic traits in plants.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://browser.planteome.org/amigo\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q15304508|P31|Q81314568\n",
      "Q15304508|P856|https://github.com/DiseaseOntology/PathogenTransmissionOntology\n",
      "Q15304508|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Teleost taxonomy ontology\"\n",
      "LAST|Den|\"An ontology covering the taxonomy of teleosts (bony fish)\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/phenoscape/teleost-taxonomy-ontology\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Uberon multi-species anatomy ontology\"\n",
      "LAST|Den|\"An integrated cross-species anatomy ontology covering animals and bridging multiple species-specific ontologies\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://uberon.org\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Units of measurement ontology\"\n",
      "LAST|Den|\"Metrical units for use in conjunction with PATO\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/bio-ontology-research-group/unit-ontology\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Unified phenotype ontology (uPheno)\"\n",
      "LAST|Den|\"The uPheno ontology integrates multiple phenotype ontologies into a unified cross-species phenotype ontology.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/obophenotype/upheno\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "Q55118646|P31|Q81314568\n",
      "Q55118646|P856|http://www.violinet.org/vaccineontology\n",
      "Q55118646|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Vertebrate trait ontology\"\n",
      "LAST|Den|\"An ontology of traits covering vertebrates\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/AnimalGenome/vertebrate-trait-ontology\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Vertebrate Taxonomy Ontology\"\n",
      "LAST|Den|\"Comprehensive hierarchy of extinct and extant vertebrate taxa.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/phenoscape/vertebrate-taxonomy-ontology\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"C. elegans Gross Anatomy Ontology\"\n",
      "LAST|Den|\"A structured controlled vocabulary of the anatomy of <i>Caenorhabditis elegans</i>.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://www.wormbase.org/\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"C. elegans development ontology\"\n",
      "LAST|Den|\"A structured controlled vocabulary of the development of <i>Caenorhabditis elegans</i>.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://www.wormbase.org/\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"C. elegans phenotype\"\n",
      "LAST|Den|\"A structured controlled vocabulary of <i>Caenorhabditis elegans</i> phenotypes\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/obophenotype/c-elegans-phenotype-ontology\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Experimental condition ontology\"\n",
      "LAST|Den|\"Conditions under which physiological and morphological measurements are made both in the clinic and in studies involving humans or model organisms.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://rgd.mcw.edu/rgdweb/ontology/view.html?acc_id=XCO:0000000\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"HUPO-PSI cross-linking and derivatization reagents controlled vocabulary\"\n",
      "LAST|Den|\"A structured controlled vocabulary for cross-linking reagents used with proteomics mass spectrometry.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|http://www.psidev.info/groups/controlled-vocabularies\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Xenopus Phenotype Ontology\"\n",
      "LAST|Den|\"An ontology of anatomical, cellular, and gene function phenotypes in Xenopus, the African clawed frogs.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/obophenotype/xenopus-phenotype-ontology\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Zebrafish Experimental Conditions Ontology\"\n",
      "LAST|Den|\"Experimental conditions applied to zebrafish, developed to facilitate experiment condition annotation at ZFIN\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/ybradford/zebrafish-experimental-conditions-ontology\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Zebrafish developmental stages ontology\"\n",
      "LAST|Den|\"Developmental stages of the Zebrafish\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://wiki.zfin.org/display/general/Anatomy+Atlases+and+Resources\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Zebrafish Phenotype Ontology\"\n",
      "LAST|Den|\"The Zebrafish Phenotype Ontology formally defines all phenotypes of the Zebrafish model organism.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/obophenotype/zebrafish-phenotype-ontology\n",
      "LAST|P361|Q7072326\n",
      "\n",
      "CREATE\n",
      "LAST|Len|\"Confidence Information Ontology\"\n",
      "LAST|Den|\"An ontology to capture confidence information about annotations.\"\n",
      "LAST|P31|Q81314568\n",
      "LAST|P856|https://github.com/BgeeDB/confidence-information-ontology\n",
      "LAST|P361|Q7072326\n",
      "\n"
     ]
    },
    {
     "ename": "TypeError",
     "evalue": "must be str, not float",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-91-74e7ac646aa2>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     13\u001b[0m             print( wid + '|P31|' + \"Q81314568\"  + '\\n' +\n\u001b[1;32m     14\u001b[0m             \u001b[0mwid\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m'|P856|'\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mrow\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'homepage'\u001b[0m\u001b[0;34m]\u001b[0m  \u001b[0;34m+\u001b[0m \u001b[0;34m'\\n'\u001b[0m \u001b[0;34m+\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 15\u001b[0;31m             wid + \"|P361|\" + \"Q7072326\" + '\\n') \n\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m: must be str, not float"
     ]
    }
   ],
   "source": [
    "for index, row in obo_df_active_selection.iterrows():\n",
    "        if row['wikidata_ids'] == 'donotexist' :\n",
    "            print(\"CREATE\\n\" + \n",
    "                  'LAST|Len|' + '\"' + row['title'] + '\"\\n' +\n",
    "                  'LAST|Den|' + '\"'+ row['description'] + '\"\\n' +\n",
    "                  'LAST|P31|' + \"Q81314568\"  + '\\n' +\n",
    "                  'LAST|P856|' + '\"' + row['homepage']  + '\"' + '\\n' +\n",
    "                  \"LAST|P361|\" + \"Q7072326\" + '\\n')\n",
    "            \n",
    "        else :\n",
    "            wid = row['wikidata_ids']\n",
    "                \n",
    "            print( wid + '|P31|' + \"Q81314568\"  + '\\n' +\n",
    "            'LAST|P856|' + '\"' + row['homepage']  + '\"' + '\\n' +\n",
    "            wid + \"|P361|\" + \"Q7072326\" + '\\n') "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
